lib/Hakyll/Core/Routes.hs

```diff
@@ -182,27 +240,62 @@ gsubRoute pattern replacement = customRoute $
 
 
 --------------------------------------------------------------------------------
--- | Get access to the metadata in order to determine the route
-metadataRoute :: (Metadata -> Routes) -> Routes
+{- | Wrapper function around other route construction functions to get
+access to the metadata (of the underlying item being processed) and use that for the
+output filepath construction.
+Warning: you have to__ensure that the accessed metadata fields actually exists__.
+
+=== __Examples__
+__Route that uses a custom slug markdown metadata field__
+
+If we want more control over the url/filepath for search engine optimization (SEO),
+we can introduce a "slug" (a common name for a unique part of an URL that is well readable by search engines and humans)
+metadata field to our (markdown) files like in the following example: 'posts\/hakyll.md'
```

Comment:
This is a mouthful :) Would it be possible to extract some of the parenthesized notes into separate sentences, or drop them completely?

Reply:
You're right. I rephrased and added an explanation link to Wikipedia